pkg/contexts/credentials/repositories/memory/config/type.go

```diff
@@ -83,7 +83,10 @@ func (a *Config) ApplyTo(ctx cfgcpi.Context, target interface{}) error {
 		return fmt.Errorf("unable to get repository for spec: %w", err)
 	}
 
-	mem := repo.(*memory.Repository)
+	mem, ok := repo.(*memory.Repository)
```

Comment:
Even here, the flow explicitly creates a memory.Repository by a generic function getting a spec for such a repository.
So, basically it must be of this type.

But you are right, these are typical *famous last words* :-)